spec/helpers/dashboard_helper_spec.rb

```diff
@@ -0,0 +1,677 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe DashboardHelper do
```

Comment:
the meat of the logic is in the helper class, so this is where most of the spec code is.